src/pages/home/report/EmojiPickerMenu.js

```diff
@@ -0,0 +1,111 @@
+import React, {Component} from 'react';
+import {View, FlatList, Text} from 'react-native';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import styles from '../../../styles/styles';
+import themeColors from '../../../styles/themes/default';
+import emojis from '../../../../assets/emojis';
+import EmojiPickerMenuItem from './EmojiPickerMenuItem';
+import TextInputFocusable from '../../../components/TextInputFocusable';
+
+const propTypes = {
+    // Controls the visibility of this component.
+    isVisible: PropTypes.bool,
+    addEmojiToTextBox: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    isVisible: false,
+};
+
+class EmojiPickerMenu extends Component {
+    constructor(props) {
+        super(props);
+
+        this.filterEmojis = _.debounce(this.filterEmojis.bind(this), 500, false);
+        this.state = {
+            filteredEmojis: emojis,
+            headerIndices: [0, 34, 60, 88, 99, 121, 148],
+        };
+    }
+
+    componentDidUpdate() {
+        if (this.props.isVisible) {
+            this.searchInput.focus();
+        }
+    }
+
+    filterEmojis(searchTerm) {
+        const normalizedSearchTerm = searchTerm.toLowerCase();
+        if (normalizedSearchTerm === '') {
+            this.setState({filteredEmojis: emojis, headerIndices: [0, 34, 60, 88, 99, 121, 148]});
+            return;
+        }
+        const newFilteredEmojiList = [];
+        emojis.forEach((emoji) => {
+            if (!emoji.header && emoji.code !== 'BLANK') {
+                emoji.keywords.forEach((keyword) => {
+                    if (keyword.includes(normalizedSearchTerm)) {
+                        newFilteredEmojiList.push(emoji);
+                    }
+                });
+            }
+        });
+
+        this.setState({filteredEmojis: newFilteredEmojiList, headerIndices: []});
+    }
+
+    renderItem({item}, addEmojiToTextBox) {
+        if (item.code === 'BLANK') {
+            return;
+        }
+
+        if (item.header) {
+            return (
+                <Text style={styles.emojiHeaderStyle}>
+                    {item.code}
+                </Text>
+            );
+        }
+
+        return (
+            <EmojiPickerMenuItem
+                onPress={addEmojiToTextBox}
+                emoji={item.code}
+            />
+        );
+    }
+
+    render() {
+        return (
+            this.props.isVisible && (
```

Comment:
I wonder if the `isVisible` prop is necessary ... since this is rendered in a modal anyways, could we remove the `isVisible` prop and let the visibility be controlled by the modal parent?

Reply:
I just tested not using the `isVisible` prop and it's causing a bunch of lag due to the animation of the modal. I tried looking for a way to disable animations and it doesn't exist. The closest I found was [this hack](https://github.com/react-native-modal/react-native-modal/issues/471) with animation times but it's still laggy, so I think we should keep this part as is

https://user-images.githubusercontent.com/42391420/113040205-c77c6c80-9155-11eb-8f74-a1f9d988b628.mp4

Reply:
It's actually only slow (~500ms delay) in web. ios and Desktop are both super fast so I'm thinking this might just be a bloated chrome thing. I'm going to get rid of the isVisible stuff